pandas/io/tests/parser/usecols.py

```diff
@@ -54,6 +54,10 @@ def test_usecols(self):
         expected.columns = ['foo', 'bar']
         tm.assert_frame_equal(result, expected)
 
+        # same length but usecols column doesn't exist - see gh-14671
```

Comment:
Can you put this in a separate test (eg call it `test_usecols_non_existing`)? And also add an explicit test case for the 'normal' case where the length does not match the number of columns (which already raises), and tests for both cases but using integers.